sunpy/net/dataretriever/sources/stereo.py

```diff
@@ -0,0 +1,359 @@
+                chk_var += 1
+            if x.__class__.__name__ == 'Detector' and regex.sub('', x.value).lower() in detectors:
+                chk_var += 1
+        return chk_var == 3
+
+
+class PLASTICClient(GenericClient):
+    """
+    Returns a list of URLS to PLASTIC files corresponding to value of input timerange.
+    URL source: `http://stereo-ssc.nascom.nasa.gov/data/beacon/`.
+    Parameters
+    ----------
+    timerange: sunpy.time.TimeRange
+        time range for which data is to be downloaded.
+        Example value - TimeRange('2015-12-30 00:00:00','2015-12-31 00:01:00')
+
+    Instrument: Fixed argument, case-insensitive = 'plastic'
+
+    Source: Two arguments, case-sensitive - 'STEREO_A' or 'STEREO_B'.
+
+    Examples
+    --------
+    >>> from sunpy.net import Fido
+    >>> from sunpy.net import attrs as a
+    >>> results = Fido.search(a.Time('2007/3/2 17:00:00', '2007/3/4 17:15:00'),
+        a.Instrument('plastic'), a.Source('STEREO_A'))
+    >>> print(results)
+       [<Table length=2>
+         Start Time           End Time      Source Instrument
+           str19               str19         str5     str7
+        ------------------- ------------------- ------ ----------
+        2007-03-02 00:00:00 2007-03-03 00:00:00  ahead    plastic
+        2007-03-03 00:00:00 2007-03-04 00:00:00  ahead    plastic]
+    """
+    def _get_url_for_timerange(self, timerange, **kwargs):
+        """
+        returns list of urls corresponding to given TimeRange.
+        """
+        if not timerange:
+            return []
+        source = kwargs.get('source')
+        table_ = {'STEREO_A':'ahead', 'STEREO_B':'behind'}
+        result = list()
+        if source == 'ahead':
+            START_DATE = datetime.datetime(2006, 10, 1)
+        else:
+            START_DATE = datetime.datetime(2006, 10, 12)
+        if timerange.start < START_DATE:
+            raise ValueError('Earliest date for which PLASTIC data is available is {:%Y-%m-%d}'.format(START_DATE))
+        prefix = 'http://stereo-ssc.nascom.nasa.gov/data/beacon/{source}/plastic/'
+        prefix_copy = prefix
+        source_dict = {'STEREO_A':'%Y', 'STEREO_B':'%y'}
+        prefix += source_dict[source]
+        prefix_copy += source_dict[source]
+        prefix += '/%m/ST{char}_LB_PLASTIC_%Y%m%d_'
+        prefix_copy += '/%m/ST{char}_LB_PLA_BROWSE_%Y%m%d_'
+
+        suffix = 'V{0:02d}.cdf'
+        suffix_copy = suffix
+        pattern = prefix.format(source=table_[source], char=table_[source][0].upper()) + suffix
+        pattern_copy = prefix_copy.format(source=table_[source], char=table_[source][0].upper()) + suffix
+        patterns = list()
+        patterns.append(pattern), patterns.append(pattern_copy)
+        for pattern_ in patterns:
+            url_pattern = [pattern_.format(i) for i in range(6, 12)]
+            arr = [Scraper(pattern__).filelist(timerange) for pattern__ in url_pattern]
+            [result.extend(url) for url in arr if len(url) > 0]
```

Comment:
No, ```arr``` is an array of arrays (Different arrays for different scrapers). You don't want to append an array of arrays (I told you before).  to ```result```. Maybe, I should add some comments here, that would do it.